manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -86,6 +86,20 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 		if err != nil {
 			log.G(ctx).WithError(err).Error("failed to find tasks with desired state REMOVE in task reaper init")
 		}
+
+		// Tasks that have been marked for shutdown (likley because of a service update)
+		// but have not progressed beyond TaskStatePending also need to be cleaned up.
+		// However, we want to let these tasks go through the regular history cleanup process
+		// as a part of the select loop below.
+		shutdownTasks, err := store.FindTasks(readTx, store.ByDesiredState(api.TaskStateShutdown))
+		if err != nil {
+			log.G(ctx).WithError(err).Error("failed to find tasks with desired state SHUTDOWN in task reaper init")
+		}
+		for _, t := range shutdownTasks {
```

Comment:
Non-blocking optimization:  I was wondering if it'd make sense to move this loop outside of the view?  If there are a lot of shutdown tasks, this could block whatever lock the view function is holding.

Reply:
also a great point.